content_script.js

```diff
@@ -1,82 +1,213 @@
+     *    is a string that uniquely identifies an iframe, the serialized contents of which should be placed at that
+     *    index of this.html.
+     */
+    this.frameHoles = {};
+
+    /**
+     * @public {Array<number>} Each number in this.styleIndices corresponds to an index in this.html at which a
+     *    serialized style attribute is located. This is because there are styles that contain quotation marks.
+     *    Because any given document being serialized could be an iframe which is nested 1 or more levels into
+     *    the root document, the exact quotes that will be used must be determined when the frames are being put
+     *    together, so that they can be properly escaped.
+     */
+    this.styleIndices = [];
   }
-  else {
-    html.push(new Array(depth+1).join("  "));
-    html.push("<" + element.tagName + " ");
-
-    var style = element.ownerDocument.defaultView.getComputedStyle(element, null).cssText;
-    style = style.replace(/"/g, "'");
-    html.push('style="' + style + '" ');
-
-    var attributes = element.attributes;
-    if (attributes) {
-      for (var i = 0; i < attributes.length; i++) {
-        var attribute = attributes[i];
-        if (attribute.name.toLowerCase() == "src") {
-          html.push(attribute.name + "=");
-          holes[html.length] = attribute.value; // mark location to insert url later
-          html.push(""); // create filler entry
+
+  /**
+   * Takes an html element, and populates this object's fields such that it can eventually be converted into an
+   * html text file.
+   *
+   * @param {Element} element The Element to serialize
+   * @param {number} depth The number of parents this element has in the current frame
+   * @private
+   */ 
+  serializeTree(element, depth) {
+    if (!element.tagName && element.nodeType != Node.TEXT_NODE) {
+      // ignore elements that don't have tags and are not text
+    } else if (element.tagName && this.FILTERED_TAGS.has(element.tagName.toLowerCase())) {
+      // filter out elements that are in filteredTags
+    } else if (element.nodeType == Node.TEXT_NODE) {
+      this.html.push(element.textContent);
+    } else {
+      this.html.push(new Array(depth+1).join('  '));
+      this.html.push(`<${element.tagName.toLowerCase()} `);
+
+      var style = element.ownerDocument.defaultView.getComputedStyle(element, null).cssText;
+      this.styleIndices.push(this.html.length);
+      this.html.push(`style="${style}" `);
+
+      var attributes = element.attributes;
+      if (attributes) {
+        for (var i = 0, attribute; attribute = attributes[i]; i++) {
+          if (attribute.name.toLowerCase() == 'src') {
+            if (element.tagName.toLowerCase() != 'iframe') {
+              this.html.push(`${attribute.name}=`);
+              this.html.push(''); // entry where properly escaped quotes will go
+              this.srcHoles[this.html.length] = attribute.value; // mark location to insert url later
```

Comment:
It looks like  you're only making one hole, but pushing three different placeholder entries in the html array?